### PR TITLE
[release-3.5] server,test: refresh cache on each NewAuthStore

### PR DESCRIPTION
**Pull request by vivekpatani**
- permissions were incorrectly loaded on restarts.
- #14355
- Backport of https://github.com/etcd-io/etcd/pull/14358

Signed-off-by: vivekpatani <9080894+vivekpatani@users.noreply.github.com>

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by mitake** [comment]
`TestV3AuthRestartMember` is also failing only in linux-amd64-integration-2-cpu. Let me check.

**Comment by mitake** [comment]
LGTM, the newly added test is a little bit flaky though. If it keeps failing, I think it's valuable to revisit the timeout config.
Minor request: could you revisit the email address @vivekpatani ? The github one is detected as unacceptable email address.
Defer to @ahrtr

**Comment by ahrtr** [comment]
Please signoff the commit

**Comment by serathius** [comment]
This blocks v3.5.5 release. @vivekpatani are you planning to work on this?

**Comment by vivekpatani** [comment]
@serathius yes, there is an issue with a flaky test. Sorry, was OoO.

@ahrtr @mitake I'm not able to reproduce the issue locally (or inside docker). I can increase the timeout, but I'll have to create another PR to fix that, and then rebase this once that one is merged. But given the fact, I can't reproduce the issue correctly, I'm not sure what timeout needs to be set to. Any recommendations?

@ahrtr I'll also look at fixing the signature. The github one is what `github.com` provides for privacy reasons but let me fix that.

**Comment by serathius** [comment]
All tests pass apart of DCO, please sign all commits by running `git rebase origin/release-3.5 --signoff`, and push the HEAD with force `git push origin HEAD:release-3.5 -f`

**Comment by vivekpatani** [comment]
@serathius I think there's a flaky test, but I've re pushed the fix for DCO.

**Review by ahrtr — APPROVED** [review]
LGTM  

Thank you @vivekpatani 

defer to @mitake to approve & merge this PR.

**Comment by ahrtr** [comment]
> But given the fact, I can't reproduce the issue correctly, I'm not sure what timeout needs to be set to. Any recommendations?

Let's investigate the test failure in https://github.com/etcd-io/etcd/pull/14410

**Comment by mitake** [comment]
LGTM, thanks a lot @vivekpatani . Let’s discuss about the flaky test in https://github.com/etcd-io/etcd/pull/14410

**Pull request merged**
